src/hasi/segmentation_pipeline.py

```diff
@@ -0,0 +1,350 @@
+    parameter_object.SetParameter("DefaultPixelValue", "-1024")
+    parameter_object.SetParameter("Metric", "AdvancedMeanSquares")
+    # parameter_object.SetParameter("FixedInternalImagePixelType", "short")
+    # parameter_object.SetParameter("MovingInternalImagePixelType", "short")
+    # we still have to use float pixels
+
+    print('Starting atlas registration')
+    registered, elastix_transform = itk.elastix_registration_method(
+        case_bone_image.astype(itk.F),  # fixed image is used as primary input to the filter
+        moving_image=atlas_aa_image.astype(itk.F),
+        # moving_mask=atlas_aa_segmentation,
+        parameter_object=parameter_object,
+        initial_transform_parameter_file_name=atlas_to_case_filename + '.txt',
+        # log_to_console=True,
+        output_directory=root_dir + bone + '/',
+        log_file_name=case + '-' + atlas + '-elx.log'
+    )
+
+    # serialize each parameter map to a file.
+    for index in range(elastix_transform.GetNumberOfParameterMaps()):
+        parameter_map = elastix_transform.GetParameterMap(index)
+        elastix_transform.WriteParameterFile(
+            parameter_map,
+            case_base + f".{index}.txt")
+
+    registered_filename = case_base + '-reg.nrrd'
+    print(f'Writing registered image to file {registered_filename}')
+    itk.imwrite(registered.astype(itk.SS), registered_filename)
+
+    print('Running transformix')
+    elastix_transform.SetParameter('FinalBSplineInterpolationOrder', '0')
+    result_image_transformix = itk.transformix_filter(
+        atlas_aa_segmentation.astype(itk.F),  # transformix only works with float pixels
+        elastix_transform,
+        # reference image?
+    )
+    result_image = result_image_transformix.astype(itk.UC)
+    registered_label_file = case_base + '-label.nrrd'
+    print(f'Writing deformed atlas to {registered_label_file}')
+    itk.imwrite(result_image, registered_label_file, compression=True)
+
+    print('Computing morphometry features')
+    morphometry_filter = itk.BoneMorphometryFeaturesFilter[type(atlas_aa_image)].New(case_bone_image)
+    label_names = {
+        1: 'Diaphysis',
+        2: 'Metaphysis',
+        3: 'Growth Plate',
+        4: 'Epiphysis',
+    }
+    # TODO: add proper reading of labels from .seg.nrrd files so this is not hardcoded
+    # label_names = {
+    #     1: 'Cortical',
+    #     2: 'Trabecular VOI',
+    #     3: 'New Trabecular VOI',
+    #     4: 'New Cortical VOI',
+    # }
+    for label in label_names:
+        print(f'Label {label_names[label]}')
+        label_region = itk.binary_threshold_image_filter(
+            result_image, lower_threshold=label, upper_threshold=label)
+        morphometry_filter.SetMaskImage(label_region)
+        morphometry_filter.Update()
+
+        csv_filename = root_dir + bone + '/' + atlas + '-BoneMorphometry.csv'
+        with open(csv_filename, 'a') as morphometry_csv:
+            # Bone,Atlas,Case,Label,BVTV,TbN,TbTh,TbSp,BSBV
```

Comment:
Would be helpful to reference [itkBoneMorphometryFeaturesFilter.h](https://github.com/InsightSoftwareConsortium/ITKBoneMorphometry/blob/master/include/itkBoneMorphometryFeaturesFilter.h#L35) in a comment here for feature names

Reply:
Will do.